app/controllers/courses/sites_controller.rb

```diff
@@ -8,6 +8,28 @@ class SitesController < ApplicationController
 
     def edit; end
 
+    def update
+      @course.provider_code = @provider.provider_code
+      @course.sites = params.dig(:course, :site_statuses_attributes)
+        &.values
+        &.select { |field| field["selected"] == "1" }
+        &.map { |field| @provider.sites.find { |site| site.id == field["id"] } }
```

Comment:
Paging @misaka for a way to make this more 整頓

Reply:
Ya, not sure. For one thing, I'm not sure about the safeties ... do you really want to assign `@course.sites` to nil? Aside from that, it might be nice to use the `where` with a list of ids.

```ruby
      selected_site_ids = params.dig(:course, :site_statuses_attributes).values
        .select { |f| f['selected'] == '1' }
        .map    { |f| f['id'].to_i }
      @course.sites = @provider.sites.where(id: selected_site_ids)
```

Reply:
No `.where` because `@provider.sites` is a plain array, not an AR model. Have improved it with the rest